### PR TITLE
test: cypress - updated dateColumn test from js to ts

### PR DESCRIPTION
**Pull request by NandanAnantharamu**
Updated Flaky tests Date_column_editing_1_spec.js and Date_column_editing_2_spec.js to ts.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  agHelper,
+  locators,
+  propPane,
+  draggableWidgets,
+  assertHelper,
+} from "../../../../../support/Objects/ObjectsCore";
+
+describe("Table widget date column inline editing functionality", () => {
+  before(() => {
+    agHelper.AddDsl("Table/DateCellEditingDSL");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:13]: Can we DragDrop date widget & proceed. Is DSL really needed here?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        table._lastChildDatePicker
+      }`,
+      -1,
+      "{enter}",
+    );
+  });
+
+  it("4. should allow ISO 8601 format date and not throw a disallowed validation error", () => {
+    entityExplorer.SelectEntityByName("Table1");
+    agHelper.GetNClick(propPane._goBackToProperty);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:129]: Revert private for this locator back & use `propPane.NavigateBackToPropertyPane`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+  it("7. should check property pane Required toggle functionality", () => {
+    entityExplorer.SelectEntityByName("Table1");
+    agHelper.GetNClick(propPane._goBackToProperty);
+    table.EditColumn("release_date", "v2");
+    propPane.TogglePropertyState("Required", "On");
+    assertHelper.AssertNetworkStatus("updateLayout", 200);
+    agHelper.Sleep(2000);
+    agHelper.HoverElement(`${table._tableDataNthChild}`);
+    agHelper.GetNClick(table._editCellIconDiv, 0, true);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:165]: `ClickOnEditIcon()` was used here to fix it. Why is it replaced with direct click here?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  it("2. should check visible property control functionality", () => {
+    entityExplorer.SelectEntityByName("Table1");
+    table.EditColumn("release_date", "v2");
+    propPane.TogglePropertyState("Visible", "Off");
+    agHelper.Sleep(1000);
+    agHelper.AssertElementExist(
+      `${table._tableV2Head} ${table._columnHeaderDiv("release_date")} ${
+        table._hiddenHeader
+      }`,
+    );
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:69]: Use - `AssertHiddenColumns` + remove created locators.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts (hunk 1 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:12]: Why is DSL needed?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  entityExplorer,
+  agHelper,
+  locators,
+  propPane,
+  assertHelper,
+} from "../../../../../support/Objects/ObjectsCore";
+
+describe("Table widget date column inline editing functionality", () => {
+  before(() => {
+    agHelper.AddDsl("Table/DateCellEditingDSL");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:12]: Why no DragDrop?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+    );
+    entityExplorer.SelectEntityByName("Table1");
+    propPane.NavigateBackToPropertyPane();
+    table.EditColumn("release_date", "v2");
+    propPane.TogglePropertyState("Visible", "On");
+    agHelper.Sleep(1000);
+    agHelper.AssertElementExist(
+      `${table._tableV2Head} ${table._columnHeaderDiv("release_date")} ${
+        table._draggableHeader
+      }`,
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:78]: Use `table.AssertTableHeaderOrder`

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+      -1,
+      "{enter}",
+    );
+  });
+
+  it("4. should allow ISO 8601 format date and not throw a disallowed validation error", () => {
+    entityExplorer.SelectEntityByName("Table1");
+    propPane.NavigateBackToPropertyPane();
+    agHelper.UpdateCodeInput(
+      propPane._propertyControl("tabledata"),
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:129]: Use` Table data` - as seen in UI.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+      '[{ "dateValue": "2023-02-02T13:39:38.367857Z" }]',
+    );
+    agHelper.Sleep(3000);
+    table.ChangeColumnType("dateValue", "Date", "v2");
+    table.EditColumn("dateValue", "v2");
+    propPane.SelectPropertiesDropDown("dateformat", "ISO 8601");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:135]: `dateformat` - Use this value as seen in UI.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    // we should not see an error after selecting the ISO 8061 format
+    agHelper.AssertElementAbsence(
+      `${propPane._propertyDateFormat} ${table._codeMirrorError}`,
+    );
+    propPane.ToggleJSMode("dateformat", true);
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:140]: `dateformat` - As seen in UI.

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:140]: + all places.

File: app/client/cypress/support/Pages/AggregateHelper.ts [hunk 10]
```diff
@@ -1422,6 +1418,16 @@ export class AggregateHelper extends ReusableHelper {
       apiToValidate && this.assertHelper.AssertNetworkStatus(apiToValidate);
   }
 
+  public AssertValue(
```
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AggregateHelper.ts:1454]: Why is AssertText not used even after review but this comment in resolved before that? What  reason?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+    propPane.NavigateBackToPropertyPane();
+    table.EditColumn("release_date", "v2");
+    agHelper.GetNClick(
+      `${locators._propertyControl}displayformat ${table._showArrow}`,
+    );
+    agHelper
+      .GetElement(locators._dropdownText)
+      .children()
+      .contains("Do MMM YYYY")
+      .click();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:95]: Why is `dataSources.SelectPropertiesDropDown` not used?

Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: There was another dropdown method which i have used and udpated

Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: Also Assertvalue is removed i have used GetText and done validation

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 9 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 8 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 7 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 6 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts]: This was not working for me , the header text was updated properly for assertion, however i have added a method to  validate that header visbility on the same lines as hidden.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 5 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts:12]: I cant make out the pre-requisites hence we are using DSL approach here

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+      table._popoverContent,
+      "This field is required",
+    );
+    agHelper.GetNClick(
+      `${table._dateInputPopover} [aria-label='Wed May 26 2021']`,
+    );
+    agHelper.AssertElementAbsence(table._popoverContent);
+  });
+
+  it("7. should check date cells behave as expected when adding a new row to table", () => {
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:159]: previous review comment u had asked me to combine 2 tests hence its 7

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+    );
+    table.ClickOnEditIcon(0, 2);
+    agHelper.GetNClick(
+      `${table._dateInputPopover} [aria-label='Wed May 26 2021']`,
+    );
+    agHelper.AssertElementExist(table._popoverContent);
+    agHelper.GetNAssertContains(
+      table._popoverContent,
+      "This field is required",
+    );
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+    propPane.TogglePropertyState("Required", "On");
+    agHelper.Sleep(2000);
+    table.ClickOnEditIcon(0, 2);
+    agHelper.GetNClick(
+      `${table._dateInputPopover} [aria-label='Wed May 26 2021']`,
+    );
+    table.ClickOnEditIcon(0, 2);
+    agHelper.GetNClick(
+      `${table._dateInputPopover} [aria-label='Wed May 26 2021']`,
+    );
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:147]: there is a use-case for which it had to be clicked twice hence its twice, there is a error validation followed by this step

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+    agHelper.AssertElementAbsence(`${table._dateRangePicker}`);
+    entityExplorer.SelectEntityByName("Table1");
+    propPane.NavigateBackToPropertyPane();
+    table.EditColumn("release_date", "v2");
+    propPane.TogglePropertyState("Show Shortcuts", "On");
+    agHelper.Sleep(2000);
+    agHelper.GetElement(`${table._tableNthChild}`).dblclick({
+      force: true,
+    });
+    agHelper.AssertElementExist(`${table._dateRangePicker}`);
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: done

File: app/client/cypress/support/Pages/Table.ts [hunk 16]
```diff
@@ -157,6 +156,35 @@ export class Table {
...
+  _tableNthChild =
+    ".t--draggable-tablewidgetv2 .tbody .tr:nth-child(1) div:nth-child(3)";
+  _hiddenHeader = ".hidden-header";
+  _draggableHeader = " .draggable-header";
+  _lastChildDatePicker = "div:last-child .react-datepicker-wrapper";
+  _codeMirrorError = ".t--codemirror-has-error";
+  _canvasWidgetType = "[type='CANVAS_WIDGET']";
+  _showArrow = ".rc-select-show-arrow";
+  _codeEditorWrapper = ".t--code-editor-wrapper";
+  _dateRangePicker = ".bp3-dateinput-popover .bp3-daterangepicker-shortcuts";
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts]: updated the locator name

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 17]
```diff
@@ -0,0 +1,210 @@
...
+    agHelper.AssertElementExist(table._editCellEditor);
+    agHelper.GetNClick(`${locators._widgetInCanvas(draggableWidgets.TEXT)}`, 0);
+    agHelper.AssertElementAbsence(table._dateInputPopover);
+    agHelper.AssertElementAbsence(table._editCellEditor);
+    agHelper.GetNAssertContains(
+      `${table._canvasWidgetType} ${locators._widgetInDeployed(
+        draggableWidgets.TEXT,
+      )} ${locators._textWidget}`,
+      `{"revenue":42600000,"imdb_id":"tt3228774","release_date":"2021-05-17"}`,
+    );
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: done

File: app/client/cypress/support/Pages/Table.ts [hunk 18]
```diff
@@ -157,6 +156,35 @@ export class Table {
...
+  _datePicker = ".bp3-datepicker";
+  _dayPickerWeek = ".bp3-datepicker .DayPicker .DayPicker-Body .DayPicker-Week";
+  _timePickerHour = ".bp3-timepicker-input-row .bp3-timepicker-hour";
+  _timePickerMinute = ".bp3-timepicker-input-row .bp3-timepicker-minute";
+  _timePickerSecond = ".bp3-timepicker-input-row .bp3-timepicker-second";
+  _timePickerRow = ".bp3-timepicker-input-row";
+  _tableV2Head = ".t--draggable-tablewidgetv2 .thead";
+  _timeprecisionPopover =
+    ".t--property-control-timeprecision .bp3-popover-target";
+  _tableNthChild =
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/support/Pages/Table.ts]: updated

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+    },
+  );
+
+  it("2. should check that user can edit date in table cell", () => {
+    agHelper.GetElement(`${table._tableNthChild}`).dblclick();
+    agHelper.AssertElementExist(table._dateInputPopover);
+    agHelper.AssertElementExist(table._editCellEditor);
+    agHelper.GetNClick(
+      `${table._dateInputPopover} [aria-label='Mon May 17 2021']`,
+    );
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:40]: removed the duplicate part

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+    "1. should check that edit check box is enabled for date type column in the columns list" +
+      "and check that date cell edit mode can be turned on",
+    () => {
+      entityExplorer.SelectEntityByName("Table1");
+      agHelper.AssertElementEnabledDisabled(
+        table._columnCheckbox("release_date"),
+        0,
+        false,
+      );
+      entityExplorer.SelectEntityByName("Table1");
```
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts]: removed

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts (hunk 4 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_2_spec.ts]: done

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:159]: What cases from current spec are combined in this?

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts (hunk 1 above)
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:12]: I cant make out the pre-requisites hence we are using DSL approach here

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:159]: We had 2 tests like this "1. should check that edit check box is enabled for date type column in the columns list" 
      "2. And check that date cell edit mode can be turned on". now 1 and 2 are merged as 1

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts (hunk 14 above)
[the remark this answers is not in the document]
Comment (NandanAnantharamu) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:147]: - It selects a date while editing first time
- it unselects the date selected on previous occasion by clicking edit again
- And the following step evaluates the error message displayed

[the remark this answers is not in the document]
Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/TableV2/Date_column_editing_1_spec.ts:147]: Got it, please comment the same in spec against each click - so it makes it more readable about why 2 clicks are needed.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**